src/GafferSceneUI/ShaderView.cpp

```diff
@@ -275,6 +283,12 @@ void ShaderView::sceneRegistrationChanged( const PrefixAndName &prefixAndName )
 	}
 }
 
+void ShaderView::rendererRegistrationChanged()
+{
+	m_registryUpdated = true;
```

Comment:
Since `m_registryUpdated` breaks ABI, perhaps we could do `m_rendererShaderPrefix = ""` or `m_renderer = nullptr` (or both) here instead, since it seems like that would also force a refresh in `updateRenderer()`?

Reply:
`m_renderer = nullptr` is making gaffer freeze that's why I used the registry var but I didn't think about the prefix which does the same job. I will remove `m_registryUpdated` and simply set `m_rendererShaderPrefix = ""`.